### PR TITLE
Pocketbook: Try to auto-detect HW feature caps more precisely

### PR DESCRIPTION
**Pull request by ezdiy**
The differences between Allwinner/MXC matter surprisingly little in practice (https://github.com/koreader/koreader-base/issues/1202), it's not even worth it to bother the base mxc driver with it. Instead, frontend can make some informed choice by probing for the hardware and tweaking requested device caps accordingly.

On Allwinner B288, we disable both hwrot (done in software in kernel and terribly, ie not worth it) and hwinverse (no support whatsoever). On NXT we'll keep both on, as the kernel driver seems to be [using the hardware properly](https://github.com/pocketbook/Platform_MX6/blob/master/Kernel_631/drivers/video/mxc/mxc_epdc_fb.c#L2218)

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6721)

### CODE REVIEW

File: frontend/device/pocketbook/device.lua [hunk 1]
```diff
@@ -37,8 +37,10 @@ local PocketBook = Generic:new{
     -- all devices that have warmth lights use inkview api
     hasNaturalLightApi = yes,
 
-    -- Apparently, HW inversion is a pipedream on PB (#6669)
-    canHWInvert = no,
+    -- NOTE: Apparently, HW inversion is a pipedream on PB (#6669), ... well, on sunxi chipsets anyway.
+    -- For which we now probe in fbinfoOverride() and tweak the flag to "no".
+    -- NXT/other chipsets *should* work (PB631), but in case it doesn't on your device, set this to "no" in here.
```
Comment (NiLuJe) [inline frontend/device/pocketbook/device.lua:42]: Nitpick: It's NTX, not NXT ;).

(And the SoCs are from NXP (ex-Freescale), which doesn't help avoid confusion ^^).

**Review by NiLuJe — APPROVED** [review]
Besides the NTX nit, LGTM ;).

Random query, though: what does `islwaysPortrait` does now on devices with `usingForcedRotation`?

**Comment by ezdiy** [comment]
> Random query, though: what does `isAlwaysPortrait` does now on devices with `usingForcedRotation`?

The flags are mutually exclusive. So even though the device desc sets *both*, the tweak decides on one, another or none. Will add some comment about too I guess.

File: frontend/device/generic/device.lua [hunk 2]
```diff
@@ -84,11 +84,11 @@ local Device = {
     isAlwaysPortrait = no,
     -- On some devices (eg newer pocketbook) we can force HW rotation on the fly (before each update)
     -- The value here is table of 4 elements mapping the sensible linux constants to whatever
-    -- nonsense the device actually has. Canonically it should be { 0, 1, 2, 3 } if the device
+    -- nonsense the device actually has. Canonically it should return { 0, 1, 2, 3 } if the device
     -- matches <linux/fb.h> FB_ROTATE_* constants.
     -- See https://github.com/koreader/koreader-base/blob/master/ffi/framebuffer.lua for full template
     -- of the table expected.
-    usingForcedRotation = nil,
+    usingForcedRotation = function() return nil end,
```
Comment (Frenzie) [inline frontend/device/generic/device.lua:91]:
```suggestion
    usingForcedRotation = function() return end,
```

**Review by Frenzie — APPROVED** [review]
Sure, lgtm

**Review by NiLuJe — APPROVED** [review]

File: frontend/device/generic/device.lua (hunk 2 above)
Comment (ezdiy) [reply frontend/device/generic/device.lua:91]: (intentional, to signal there's meant to be no table)

**Pull request merged**